include/seqan3/io/sequence_file/format_sam.hpp

```diff
@@ -0,0 +1,347 @@
+          {
+              auto constexpr is_legal_alph = is_in_alphabet<seq_legal_alph_type>;
+              ranges::copy(seq_view | view::transform([is_legal_alph] (char const c) // enforce legal alphabet
+                                      {
+                                          if (!is_legal_alph(c))
+                                          {
+                                              throw parse_error{std::string{"Encountered an unexpected letter: "} +
+                                                                  is_legal_alph.msg.string() +
+                                                                  " evaluated to false on " +
+                                                                  detail::make_printable(c)};
+                                          }
+                                          return c;
+                                      })
+                                    | view::char_to<value_type_t<seq_type>>,      // convert to actual target alphabet
+                           ranges::back_insert_iterator{sequence});
+
+              sequence_size_after = ranges::size(sequence);
+          }
+          else
+          {
+              auto it = ranges::begin(seq_view);
+              auto it_end = ranges::end(seq_view);
+              while (it != it_end)
+             {
+                  ++it;
+                  ++sequence_size_after;
+             }
+          }
+          detail::consume(stream_view | ranges::view::take_while(is_tab));
+
+          /* Qualities */
+          if constexpr (seq_qual_combined)
+          {
+              // seq_qual field implies that they are the same variable
+              assert(std::addressof(sequence) == std::addressof(qualities));
+              ranges::copy(stream_view | view::char_to<typename value_type_t<qual_type>::quality_alphabet_type>,
+                           ranges::begin(qualities) + sequence_size_before);
+          }
+          else if constexpr (!detail::decays_to_ignore_v<qual_type>)
+          {
+            ranges::copy(stream_view | view::take_until_or_throw(is_cntrl||is_tab)
+                                     | view::char_to<value_type_t<qual_type>>,
+                        ranges::back_insert_iterator{qualities});
+            // In case of tags
+            detail::consume(stream_view | ranges::view::take_while(is_tab));
+            detail::consume(stream_view | ranges::view::take_while(!is_cntrl));
+          }
+          else
+          {
+              detail::consume(stream_view | ranges::view::take_while(!is_cntrl));
+          }
+
+          // make sure "buffer at end" implies "stream at end"
+          if ((std::istreambuf_iterator<char>{stream} == std::istreambuf_iterator<char>{}) &&
+              (!stream.eof()))
+          {
+              stream.get(); // triggers error in stream and sets eof
+          }
+
+          if ((std::istreambuf_iterator<char>{stream} == std::istreambuf_iterator<char>{}) &&
```

Comment:
this code block is exactly the same as the one above (L 250-255) is it not? If so it can be removed.